sim-lib/src/lib.rs

```diff
@@ -247,27 +273,36 @@ impl Simulation {
         for payment_flow in self.activity.iter() {
             // We need every source node that is configured to execute some activity to be included in our set of
             // nodes so that we can execute actions on it.
-            let source_node =
-                self.nodes
-                    .get(&payment_flow.source)
-                    .ok_or(LightningError::ValidationError(format!(
-                        "source node not found {}",
-                        payment_flow.source,
-                    )))?;
+            let src_id = payment_flow
+                .source
+                .get_pk()
+                .map_err(LightningError::ValidationError)?;
```

Comment:
just realized these should be `SimulationError::ValidationError`

Reply:
Agreed, tho that does not exist, we may want to move it.